client/blocks/upgrade-nudge/index.jsx

```diff
@@ -100,7 +102,7 @@ export class UpgradeNudge extends React.Component {
 			( feature === FEATURE_NO_ADS && site.options.wordads ) ||
 			( ( ! jetpack && site.jetpack ) || ( jetpack && ! site.jetpack ) );
 
-		if ( shouldNotDisplay ) {
+		if ( shouldNotDisplay && ! shouldDisplay ) {
```

Comment:
There's a refactoring opportunity based on the observation that most of `UpgradeNudge`'s props are used to compute `shouldNotDisplay` and nothing else. The condition could be therefore moved to the `connect` function:

Before:
```js
function UpgradeNudge( { a, b, c, d } ) {
  const shouldDisplay = complexCondition( a, b, c, d );
  ...
}

connect( state => ( {
  a: ...
  b: ...
  c: ...
  d: ...
} ) )( UpgradeNudge );
```

After:
```js
function UpgradeNudge( { shouldDisplay } ) {
  ...
}

connect( state => ( {
  shouldDisplay: complexCondition( a, b, c, d ),
} )( UpgradeNudge );
```

That makes the component more universal (has fewer props, `shouldDisplay` is usable in both devdocs and production).